test/functional/mempool_persist.py

```diff
@@ -66,16 +66,17 @@ def run_test(self):
 
         self.log.debug("Stop-start the nodes. Verify that node0 has the transactions in its mempool and node1 does not. Verify that node2 calculates its balance correctly after loading wallet transactions.")
         self.stop_nodes()
+        self.start_node(1)  # Give this one a head-start, so we can be "extra-sure" that it didn't load anything later
         self.start_node(0)
-        self.start_node(1)
         self.start_node(2)
         # Give bitcoind a second to reload the mempool
-        time.sleep(1)
-        wait_until(lambda: len(self.nodes[0].getrawmempool()) == 5)
-        wait_until(lambda: len(self.nodes[2].getrawmempool()) == 5)
+        wait_until(lambda: len(self.nodes[0].getrawmempool()) == 5, timeout=1)
```

Comment:
Use sync_mempool with nodes 0 and 2?

Reply:
We don't want to `sync_mempools`, but check that there are exactly 5 txs in the mempool. (Hopefully the ones that were there on shutdown)